test/relayclient/RelayClient.test.ts

```diff
@@ -194,48 +188,47 @@ contract('RelayClient', function (accounts) {
     })
 
     it('should report relays that timeout to the Known Relays Manager', async function () {
-      const badHttpClient = new BadHttpClient(false, false, true)
-      const dependencyTree = RelayClient.getDefaultDependencies(underlyingProvider, gsnConfig)
-      dependencyTree.httpClient = badHttpClient
+      const badHttpClient = new BadHttpClient(configureGSN(gsnConfig), false, false, true)
+      const dependencyTree = getDependencies(configureGSN(gsnConfig), underlyingProvider, { httpClient: badHttpClient })
       const relayClient =
-        new RelayClient(dependencyTree, relayHub.address, gsnConfig.relayClientConfig)
+        new RelayClient(underlyingProvider, gsnConfig, dependencyTree)
       // @ts-ignore (sinon allows spying on all methods of the object, but TypeScript does not seem to know that)
       sinon.spy(dependencyTree.knownRelaysManager)
       await relayClient._attemptRelay(relayInfo, optionsWithGas)
       expect(dependencyTree.knownRelaysManager.saveRelayFailure).to.have.been.calledWith(sinon.match.any, relayManager, relayUrl)
     })
 
     it('should not report relays if error is not timeout', async function () {
-      const badHttpClient = new BadHttpClient(false, true, false)
-      const dependencyTree = RelayClient.getDefaultDependencies(underlyingProvider, gsnConfig)
+      const badHttpClient = new BadHttpClient(configureGSN(gsnConfig), false, true, false)
+      const dependencyTree = getDependencies(configureGSN(gsnConfig), underlyingProvider, { httpClient: badHttpClient })
       dependencyTree.httpClient = badHttpClient
       const relayClient =
-        new RelayClient(dependencyTree, relayHub.address, gsnConfig.relayClientConfig)
+        new RelayClient(underlyingProvider, gsnConfig, dependencyTree)
       // @ts-ignore (sinon allows spying on all methods of the object, but TypeScript does not seem to know that)
       sinon.spy(dependencyTree.knownRelaysManager)
       await relayClient._attemptRelay(relayInfo, optionsWithGas)
       expect(dependencyTree.knownRelaysManager.saveRelayFailure).to.have.not.been.called
     })
 
     it('should return error if transaction returned by a relay does not pass validation', async function () {
-      const badHttpClient = new BadHttpClient(false, false, false, pingResponse, '0x123')
-      const dependencyTree = RelayClient.getDefaultDependencies(underlyingProvider, gsnConfig)
-      const badTransactionValidator = new BadRelayedTransactionValidator(true, dependencyTree.contractInteractor, gsnConfig.relayHubAddress, defaultEnvironment.chainId, gsnConfig.transactionValidatorConfig)
-      dependencyTree.httpClient = badHttpClient
-      dependencyTree.transactionValidator = badTransactionValidator
+      const badHttpClient = new BadHttpClient(configureGSN(gsnConfig), false, false, false, pingResponse, '0x123')
+      let dependencyTree = getDependencies(configureGSN(gsnConfig), underlyingProvider)
```

Comment:
thought we renamed dependencyTree ..
(```gsnDependencies``` sounds great name. better then ```overrideDependencies```)

Reply:
dependencyTree is a test var name, it does not matter. `overrideDependencies` because it is not a complete object, unlike `gsnDependencies` which is.